lbz/events/api.py

```diff
@@ -66,8 +69,9 @@ def get_all_failed_events(self) -> List[BaseEvent]:
 
     def send(self) -> None:
         try:
-            if self._pending_events:
-                entries = [self._create_eb_entry(event) for event in self._pending_events]
+            for i in range(0, len(self._pending_events), MAX_EVENTS_TO_SEND_AT_ONCE):
```

Comment:
_FYI:_ This is the real fix I have implemented, the rest of the changes are related to the repository files.

Reply:
with the try except logic it should _mark_sent for those sent in each try.
also I would move then try to within for loop.
so both mark sent and mark field should take chunks and mark only them the for loop should become a while as the pending_events will be mutating (for will fail) 
consider adding _get_next_chunk_of_events method that will utilise that limit.

Reply:
I completely agree with you, I don't know why I skipped this part 🤦